### PR TITLE
Make isCanBeDeclaredAgainst() a public API

### PR DESCRIPTION
**Pull request by tresat**
This needs to be public for consumers to use it to avoid calling methods that will emit warnings on non-declarable configurations, for instance, while using `configurations.all {}`.

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
Do you think there is a need to announce this in the release notes?  It seems kind of self-explanatory, and we don't list every new method there.

File: subprojects/core-api/src/main/java/org/gradle/api/artifacts/Configuration.java [hunk 1]
```diff
@@ -640,6 +640,24 @@ public String determineName(Configuration c) {
      */
     boolean isCanBeResolved();
 
+    /**
+     * Configures if a configuration can have dependencies declared upon it.
+     *
+     * @since 8.2
+     */
+    @Incubating
+    void setCanBeDeclaredAgainst(boolean allowed);
```
Comment (jvandort) [inline subprojects/core-api/src/main/java/org/gradle/api/artifacts/Configuration.java:649]: This should use the new naming scheme

**Review by jvandort — APPROVED** [review]

**Review by jvandort — APPROVED** [review]

**Comment by jvandort** [comment]
> Do you think there is a need to announce this in the release notes? It seems kind of self-explanatory, and we don't list every new method there.

Probably not necessary

File: subprojects/core-api/src/main/java/org/gradle/api/artifacts/Configuration.java (hunk 1 above)
Comment (tresat) [reply subprojects/core-api/src/main/java/org/gradle/api/artifacts/Configuration.java:649]: Yes.  I'll update this after the other PR merges.

**Comment by tresat** [comment]
@bot-gradle test and merge

**Comment by tresat** [comment]
@bot-gradle test and merge

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**